homeassistant/components/sonos/sensor.py

```diff
@@ -64,3 +78,33 @@ def native_value(self) -> int | None:
     def available(self) -> bool:
         """Return whether this device is available."""
         return self.speaker.available and self.speaker.power_source
+
+
+class SonosAudioInputFormatSensorEntity(SonosEntity, SensorEntity):
+    """Representation of a Sonos audio import format sensor entity."""
+
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+    _attr_icon = "mdi:import"
+    _attr_should_poll = True
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID of the sensor."""
+        return f"{self.soco.uid}-audio-format"
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"{self.speaker.zone_name} Audio Input Format"
+
+    @property
+    def native_value(self) -> str:
+        """Return the current state of the text sensor."""
+        return self.speaker.soundbar_audio_input_format
+
+    def update(self) -> None:
+        """Poll the device for the current state."""
+        self.speaker.soundbar_audio_input_format = self.soco.soundbar_audio_input_format
+
+    async def _async_poll(self) -> None:
+        """Provide a stub for required ABC method."""
```

Comment:
The battery state, which also requires polling, implements this method to perform the polling. I suppose the reason is to avoid polling as these entities are not enabled per default?

Reply:
There are two types of polling in the integration:

1. Fallback polling when subscription events fail. If subscriptions are working properly, these polling endpoints should never be called.
2. "Regular" polling where the data we need is not available in subscription event callbacks. That's what we need to use for this new sensor.

Reply:
Sorry, to clarify: this `_async_poll` method is used for the fallback polling. Since we're already always polling it's not needed.